// @framer/girishrawat.heart-react/code/final_heart.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";
import styled from "styled-components"

const CourseContainer = styled.div`
  .heart {
    width: 100px;
    height: 100px;
    background: url("https://cssanimation.rocks/images/posts/steps/heart.png")
      no-repeat;
    background-position: 0 0;
    transition: background-position 1s steps(28);
    transition-duration: 0s;
  }
  .heart.is-active {
    transition-duration: 1s;
    background-position: -2800px 0;
  }

  body {
    background: linear-gradient(135deg, #121721 0%, #000000 100%) fixed;
    color: #fff;
    font: 300 16px/1.5 "Open Sans", sans-serif;
  }

  .stage {
    position: fixed;
    top: 50%;
    left: 50%;
    -webkit-transform: translate(-50%, -50%);
    transform: translate(-50%, -50%);
  }
`;

export class heart extends React.Component<Props> {

  constructor(props){
    super(props);
    this.state = {
      condition: false
    }
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e){
    this.setState({condition: !this.state.condition})
  }

  render() {
    return (
      <CourseContainer>
        <div className="body">
          <div className="stage">
            <div onClick={this.handleClick}className={this.state.condition ? "heart is-active" : "heart"}>
            </div>
          </div>
        </div>
      </CourseContainer>
    );
  }

  // Add Framer UI for this component (in the properties panel)
  // https://framer.com/learn/docs/components#code
  static propertyControls: PropertyControls<Props> = {
    text: { type: ControlType.String, title: "Text" }
  };
}
